docs/nulecule.md

```diff
@@ -0,0 +1,242 @@
+# Nulecule file
```

Comment:
I promise you it won't rot as the spec changes infrequently and we get notified of a change. Some parts of the `nulecule.md` file also pertain to atomicapp only (ex. we don't have a [pattern] added yet for params, and we user docker:// for sources not URLs).